indexing-service/src/test/java/org/apache/druid/indexing/input/DruidSegmentReaderTest.java

```diff
@@ -80,67 +84,44 @@ public class DruidSegmentReaderTest extends NullHandlingTest
   private File segmentDirectory;
 
   private final IndexIO indexIO = TestHelper.getTestIndexIO();
+  private DimensionsSpec dimensionsSpec;
+  private List<AggregatorFactory> metrics;
+  private List<InputRow> rows;
 
   @Before
   public void setUp() throws IOException
   {
     // Write a segment with two rows in it, with columns: s (string), d (double), cnt (long), met_s (complex).
-    final IncrementalIndex incrementalIndex =
-        IndexBuilder.create()
-                    .schema(
-                        new IncrementalIndexSchema.Builder()
-                            .withDimensionsSpec(
-                                new DimensionsSpec(
-                                    ImmutableList.of(
-                                        StringDimensionSchema.create("s"),
-                                        new DoubleDimensionSchema("d")
-                                    )
-                                )
-                            )
-                            .withMetrics(
-                                new CountAggregatorFactory("cnt"),
-                                new HyperUniquesAggregatorFactory("met_s", "s")
-                            )
-                            .withRollup(false)
-                            .build()
-                    )
-                    .rows(
-                        ImmutableList.of(
-                            new MapBasedInputRow(
-                                DateTimes.of("2000"),
-                                ImmutableList.of("s", "d"),
-                                ImmutableMap.<String, Object>builder()
-                                    .put("s", "foo")
-                                    .put("d", 1.23)
-                                    .build()
-                            ),
-                            new MapBasedInputRow(
-                                DateTimes.of("2000T01"),
-                                ImmutableList.of("s", "d"),
-                                ImmutableMap.<String, Object>builder()
-                                    .put("s", "bar")
-                                    .put("d", 4.56)
-                                    .build()
-                            )
-                        )
-                    )
-                    .buildIncrementalIndex();
-
-    segmentDirectory = temporaryFolder.newFolder();
+    dimensionsSpec = new DimensionsSpec(
+        ImmutableList.of(
+            StringDimensionSchema.create("s"),
```

Comment:
This is a bit of a nit, but "s" and "str" and "stringCol" are all not too very long, but the longer ones actually convey more information on their own.  I'd suggest either using "str" and "dbl" or "strCol" and "dblCol" or "stringCol" and "doubleCol", just to be more nice to the next person who reads the code.

Reply:
Just focused on refactoring the test to be used by custom dimensions and metrics spec. Updated in the latest column.